enterprise/app/encryption/encryption.tsx

```diff
@@ -0,0 +1,347 @@
+              </div>
+            </div>
+            <div className="field-row">
+              <label htmlFor="gcpProject" className="field-label">
+                Project ID
+              </label>
+              <input
+                autoComplete="off"
+                type="text"
+                name="gcpProject"
+                onChange={this.onGCPProjectChange.bind(this)}
+                value={this.state.gcpProject}
+              />
+            </div>
+            <div className="field-row">
+              <label htmlFor="gcpKeyRing" className="field-label">
+                Key Ring
+              </label>
+              <input
+                autoComplete="off"
+                type="text"
+                name="gcpKeyRing"
+                onChange={this.onGCPKeyRingChange.bind(this)}
+                value={this.state.gcpKeyRing}
+              />
+            </div>
+            <div className="field-row">
+              <label htmlFor="gcpLocation" className="field-label">
+                Key Ring Location
+              </label>
+              <input
+                autoComplete="off"
+                type="text"
+                name="gcpLocation"
+                onChange={this.onGCPLocationChange.bind(this)}
+                value={this.state.gcpLocation}
+              />
+            </div>
+            <div className="field-row">
+              <label htmlFor="gcpKey" className="field-label">
+                Key
+              </label>
+              <input
+                autoComplete="off"
+                type="text"
+                name="gcpKey"
+                onChange={this.onGCPKeyChange.bind(this)}
+                value={this.state.gcpKey}
+              />
+            </div>
+          </>
+        );
+    }
+  }
+
+  private renderKMSOptions(kms: encryption.KMS) {
+    return (
+      <div className="kms-config-row">
+        <div>
+          <input type="radio" checked={this.state.selectedKMS == kms} onChange={this.onSelectKMS.bind(this, kms)} />
+        </div>
+        <div>
+          <div className="kms-title">{this.kmsName(kms)}</div>
+          {this.state.selectedKMS == kms && <div className="kms-fields">{this.renderKMSFields(kms)}</div>}
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <>
+        {!this.state.encryptionSupported && (
+          <div>
+            <p>Encryption requires dedicated storage.</p>
+            {/*DO NOT SUBMIT: siggi to fill in wording here*/}
+            <p>Please reach out to XXX.</p>
+          </div>
+        )}
+        {this.state.encryptionSupported && this.state.encryptionEnabled && (
+          <div>
+            <p>Encryption is enabled.</p>
+
+            <FilledButton className="destructive" onClick={this.onClickDisable.bind(this)}>
+              Disable
+            </FilledButton>
+            <Modal isOpen={this.state.isDisableModalOpen} onRequestClose={this.onCloseDisableModal.bind(this)}>
```

Comment:
nit: there is also `<SimpleModalDialog>` which requires less markup and takes care of the common boilerplate around keeping the button disabled while the RPC is in progress, etc.

Feel free to ignore, since this is also fine.